keras_cv/layers/object_detection/retina_net_label_encoder.py

```diff
@@ -102,42 +107,56 @@ def _compute_box_target(self, anchor_boxes, matched_gt_boxes):
             ],
             axis=-1,
         )
-        box_target = box_target / self._box_variance
+        box_target = box_target / self.box_variance
         return box_target
 
     def _encode_sample(self, gt_boxes, anchor_boxes):
         """Creates box and classification targets for a single sample"""
         cls_ids = gt_boxes[:, 4]
         gt_boxes = gt_boxes[:, :4]
-        cls_ids = tf.cast(cls_ids, dtype=tf.float32)
-
+        cls_ids = tf.cast(cls_ids, dtype == self.dtype)
         matched_gt_idx, positive_mask, ignore_mask = self._match_anchor_boxes(
             anchor_boxes, gt_boxes
         )
         matched_gt_boxes = tf.gather(gt_boxes, matched_gt_idx)
         box_target = self._compute_box_target(anchor_boxes, matched_gt_boxes)
         matched_gt_cls_ids = tf.gather(cls_ids, matched_gt_idx)
         cls_target = tf.where(
-            tf.not_equal(positive_mask, 1.0), -1.0, matched_gt_cls_ids
+            tf.not_equal(positive_mask, 1.0), self.background_class, matched_gt_cls_ids
         )
-        cls_target = tf.where(tf.equal(ignore_mask, 1.0), -2.0, cls_target)
+        cls_target = tf.where(tf.equal(ignore_mask, 1.0), self.ignore_class, cls_target)
         cls_target = tf.expand_dims(cls_target, axis=-1)
         label = tf.concat([box_target, cls_target], axis=-1)
         return label
 
-    def call(self, images, boxes):
+    def call(self, images, target_box):
```

Comment:
This code seems like it should break. Is this meant to be called `target_boxes`?